src/main/java/edu/harvard/iq/dataverse/AuxiliaryFileServiceBean.java

```diff
@@ -119,6 +134,35 @@ public AuxiliaryFile lookupAuxiliaryFile(DataFile dataFile, String formatTag, St
             return null;
         }
     }
+    
+    @SuppressWarnings("unchecked")
+    public List<AuxiliaryFile> listAuxiliaryFiles(DataFile dataFile, String origin) {
+        
+        Query query = em.createQuery("select object(o) from AuxiliaryFile as o where o.dataFile.id = :dataFileId and o.origin = :origin");
```

Comment:
My understanding is that we prefer our queries like the one below (typed and with the query added to the entity):

    TypedQuery query = em.createNamedQuery("AuxiliaryFile.findAuxiliaryFiles", AuxiliaryFile.class);

Reply:
Changed.

Reply:
Yay, TypedQuery. Thanks.